server/src/main/java/org/elasticsearch/common/time/JavaDateFormatter.java

```diff
@@ -122,30 +115,50 @@ public TemporalAccessor parse(String input) {
         }
 
         try {
-            return parser.parse(input);
+            return doParse(input);
```

Comment:
I think a huge comment here would be warranted what happens here and why?

Reply:
absolutely, will add